### PR TITLE
Ravinder | Development | GIDK-739 | Datepicker Scroll issue

### PR DESCRIPTION
**Pull request by ravinderwalkover**
Ravinder | Development | GIDK-739 | Datepicker Scroll issue

### CODE REVIEW

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.ts [hunk 1]
```diff
@@ -98,6 +99,8 @@ export interface DateRangeClicked {
 })
 
 export class NgxDaterangepickerComponent implements OnInit, OnDestroy, OnChanges {
+    @ViewChild(CdkVirtualScrollViewport) virtualScroll: CdkVirtualScrollViewport;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.ts:102]: Missing JS Doc

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.ts [hunk 2]
```diff
@@ -354,16 +348,25 @@ export class NgxDaterangepickerComponent implements OnInit, OnDestroy, OnChanges
...
 
         if (this.inputStartDate) {
             this.startDate = this.inputStartDate;
         }
+
         if (this.inputEndDate) {
             this.endDate = this.inputEndDate;
         }
+
+        this.scrollDispatcher.scrolled().pipe(filter(event => this.virtualScroll.measureScrollOffset('top') === 0)).subscribe(event => {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.ts:363]: Please use takeUntil

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.ts [hunk 3]
```diff
@@ -354,16 +348,25 @@ export class NgxDaterangepickerComponent implements OnInit, OnDestroy, OnChanges
...
+            this.setCalendarToActiveMonth('start');
+        });
+
+        this.scrollDispatcher.scrolled().pipe(filter(event => this.virtualScroll.measureScrollOffset('bottom') === 0)).subscribe(event => {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.ts:367]: Please use takeUntil

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.html [hunk 4]
```diff
@@ -254,248 +245,262 @@ <h2>
...
-                                        (ngModelChange)="timeChanged($event, sideEnum.end)">
-                                        <option value="AM" [disabled]="timepickerVariables.end.amDisabled">AM</option>
-                                        <option value="PM" [disabled]="timepickerVariables.end.pmDisabled">PM</option>
-                                    </select>
+
+
+                                    </div>
+                                        <ng-container *ngFor="let row of calendarMonth.end.calRows">
+                                            <div class="d-flex" [class]="calendarMonth.end.classes[row].classList"
+                                                style="display:flex;">
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.html:451]: Avoid inline styles

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.html [hunk 5]
```diff
@@ -254,248 +245,262 @@ <h2>
...
+                                                    {{ calendarMonth?.end?.calendar[1][1].format(" YYYY")}}
+                                                </div>
+                                            </ng-container>
+                                            <ng-container
+                                                *ngIf="!showDropdowns || !calendarMonth.end.dropdowns">
+                                                {{this.locale.monthNames[calendarMonth?.end?.calendar[1][1].month()]}}
+                                                {{ calendarMonth?.end?.calendar[1][1].format(" YYYY")}}
+                                            </ng-container>
+                                        </div>
+                                        <div *ngIf="showWeekNumbers || showISOWeekNumbers" class="th-div">
```
[the remark this answers is not in the document]
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.html:419]: it is existing

**Pull request merged**